utils/service_discovery/abstract_config_store.py

```diff
@@ -152,11 +244,12 @@ def get_checks_to_refresh(self, identifier, **kwargs):
         if to_check:
             return to_check
         else:
-            # lastly fallback to auto_conf
-            return set(self.identifier_to_checks[self._get_image_ident(identifier)])
+            # lastly fallback to legacy name for auto-conf
+            return self.template_cache.get_check_names(self._get_image_ident(identifier))
```

Comment:
Good catch! That's a use case we didn't support before (if user-defined templates were found we skipped auto config) but I added logic for that [here](https://github.com/DataDog/dd-agent/blob/haissam/sd-cache-kv-tpls/utils/service_discovery/abstract_config_store.py#L122-L133) and forgot to update this part. Will do it now.